HandmadeMath.h

```diff
@@ -351,43 +368,81 @@ typedef union hmm_vec4
         return Elements[Index];
     }
 #endif
-} hmm_vec4;
+} HMM_Vec4;
+
+typedef union HMM_Mat2
+{
+    float Elements[2][2];
+    HMM_Vec2 Columns[2];
+
+#ifdef __cplusplus
+    inline HMM_Vec2 &operator[](const int &Index)
+    {
+        HMM_Vec2 Result;
+        float* Column = Elements[Index];
+        
+        Result.Elements[0] = Column[0];
+        Result.Elements[1] = Column[1];
+
+        return (Result);
+    }
+#endif
+} HMM_Mat2;
+    
+typedef union HMM_Mat3
+{
+    float Elements[3][3];
+    HMM_Vec3 Columns[3];
+
+#ifdef __cplusplus
+    inline HMM_Vec3 &operator[](const int &Index)
+    {
+        HMM_Vec3 Result;
+        float* Column = Elements[Index];
+        
+        Result.Elements[0] = Column[0];
+        Result.Elements[1] = Column[1];
+        Result.Elements[2] = Column[2];
+
+        return (Result);
+    }
+#endif
+} HMM_Mat3;
 
-typedef union hmm_mat4
+typedef union HMM_Mat4
 {
     float Elements[4][4];
 
-#ifdef HANDMADE_MATH__USE_SSE
-    __m128 Columns[4];
+    HMM_Vec4 Columns[4];
 
+#ifdef HANDMADE_MATH__USE_SSE
     HMM_DEPRECATED("Our matrices are column-major, so this was named incorrectly. Use Columns instead.")
     __m128 Rows[4];
```

Comment:
Let's get rid of this variable. It's been deprecated forever. (Can also just remove `HMM_DEPRECATED`.)